src/app/view/components/dot-language-selector/dot-language-selector.component.ts

```diff
@@ -12,19 +15,33 @@ export class DotLanguageSelectorComponent implements OnInit {
     @Input() value: DotLanguage;
     @Output() selected = new EventEmitter<DotLanguage>();
 
-    languagesOptions: Observable<DotLanguage[]>;
+    languagesOptions: DotLanguage[];
+    arrowDropdownlComponentSize = 32;
+    dropdownWidth: string;
 
-    constructor(private dotLanguagesService: DotLanguagesService) {}
+    constructor(private dotLanguagesService: DotLanguagesService, private stringPixels: StringPixels) {}
 
     ngOnInit() {
-        this.languagesOptions = this.dotLanguagesService.get();
+        this.dotLanguagesService
+            .get()
+            .pipe(take(1))
+            .subscribe((languages: DotLanguage[]) => {
+                this.languagesOptions = languages;
+                this.setDropdownWidth(languages);
+            });
     }
 
     /**
      * Track changes in the dropdown
      * @param {DotLanguage} language
      */
-    change(language: DotLanguage) {
+    change(language: DotLanguage): void {
         this.selected.emit(language);
     }
+
+    private setDropdownWidth(languagesOptions: DotLanguage[]): void {
+        const maxTextOption = _.maxBy(languagesOptions, (languageOption: DotLanguage) => languageOption.language.length).language;
+        const dropdownSize = this.stringPixels.getTextWidth(maxTextOption) + this.arrowDropdownlComponentSize;
+        this.dropdownWidth = `${dropdownSize}px`;
+    }
```

Comment:
the same here 

https://github.com/dotCMS/core-web/pull/673/files#diff-c6dc9be28b451d901acfd04e29d3f845R69